File: PoolBall.java
public class PoolBall extends Ball
{
private int num;

public PoolBall(double x, double y, int n)
{
super(20, x, y, 10,"images/"+ n + "ball.png");
num = n;




}

public boolean isSolid()
{
if(num >=9)
return false;
return true;
}
public int getNum()
{
return num;
}


}

File: PoolBall.kt
class PoolBall(x: Double, y: Double, val num: Int) : Ball(20.0, x, y, 10.0, "images/" + num + "ball.png") {

    val isSolid: Boolean
        get() = num < 9

    override fun checkCollisions(b: Entity?): Boolean {
        b?.let {
            return super.checkCollisions(it)
        }
        return false
    }

    override fun impact(b: Entity?) {
        if (b != null) {
            super.impact(b)
        }
    }

}